tests/unit/lib/transport/test_mqtt.py

```diff
@@ -0,0 +1,821 @@
+
+            mqtt_client.disconnect()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    @mock.patch.object(Client, 'publish')
+    def test_publish(self, mocked_publish, connect_soc):
+        """
+        Test case to test publish method of Mqtt class.
+        :return: None
+        """
+        # Encapsulate the authentication details
+        credentials = Identity(self.root_ca_cert, self.mqtt_username, self.mqtt_password,
+                               self.client_cert_file, self.client_key_file)
+
+        # Encapsulate TLS parameters
+        tls_conf = TLSConf(self.cert_required, self.tls_version, self.cipher)
+
+        # Encapsulate QoS related parameters
+        qos_details = QoSDetails(self.inflight, self.queue_size, self.retry)
+
+        mqtt_client = Mqtt(self.url, self.port, credentials, tls_conf, qos_details, self.client_id,
+                           self.client_clean_session, self.user_data, self.protocol, self.transport,
+                           self.keep_alive, self.enable_authentication, self.connection_disconnect_timeout)
+
+        # Publishing the message
+        mqtt_client.publish("test/publish", "Testing publish", 1, False)
+
+        # Check underline publish is called with correct arguments
+        mocked_publish.assert_called_with("test/publish", "Testing publish", 1, False)
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    @mock.patch.object(Client, 'subscribe')
+    @mock.patch.object(Client, "message_callback_add")
+    def test_subscribe(self, mocked_callback_add, mocked_subscribe, connect_soc):
+        """
+        Test case to test the subscribe method.
+        :return: None
+        """
+
+        # Encapsulate the authentication details
+        credentials = Identity(self.root_ca_cert, self.mqtt_username, self.mqtt_password,
+                               self.client_cert_file, self.client_key_file)
+
+        # Encapsulate TLS parameters
+        tls_conf = TLSConf(self.cert_required, self.tls_version, self.cipher)
+
+        # Encapsulate QoS related parameters
+        qos_details = QoSDetails(self.inflight, self.queue_size, self.retry)
+
+        mqtt_client = Mqtt(self.url, self.port, credentials, tls_conf, qos_details, self.client_id,
+                           self.client_clean_session, self.user_data, self.protocol, self.transport,
+                           self.keep_alive, self.enable_authentication, self.connection_disconnect_timeout)
+
+        # Subscribing to the topic
+        mqtt_client.subscribe("test/subscribe", 1, topic_subscribe_callback)
+
+        # Check underline subscribe is called with correct parameters
+        mocked_subscribe.assert_called_with("test/subscribe", 1)
+
+        # Check underline message_callback_add is called with correct parameters
+        mocked_callback_add.assert_called_with("test/subscribe", topic_subscribe_callback)
+
+    def test_QoS_class_implementation(self):
```

Comment:
Please refer 2nd answer here and try:
http://stackoverflow.com/questions/5360833/how-to-run-multiple-classes-in-single-test-suite-in-python-unit-testing

By this approach, we should have Three separate classes.
MqttTest, QoSDetailsTest, MqttMessagingAttributesTest.  These should have their own setup and teardown methods